spark/ingestion/src/test/scala/feast/ingestion/StreamingPipelineIT.scala

```diff
@@ -291,4 +296,75 @@ class StreamingPipelineIT extends SparkSpec with ForAllTestContainer {
       StreamingPipeline.createPipeline(sparkSession, configWithKafka).get
     }
   }
+
+  "Streaming pipeline" should "store valid avro messages from kafka to redis" in new Scope {
+    val avroConfig = IngestionJobConfig(
+      featureTable = FeatureTable(
+        name = "test-fs",
+        project = "default",
+        entities = Seq(Field("customer", ValueType.Enum.STRING)),
+        features = Seq(
+          Field("feature1", ValueType.Enum.INT32),
+          Field("feature2", ValueType.Enum.FLOAT)
+        )
+      ),
+      source = KafkaSource(
+        bootstrapServers = kafkaContainer.bootstrapServers,
+        topic = "avro",
+        format = AvroFormat(schemaJson = """{
+            |"type": "record",
+            |"name": "TestMessage",
+            |"fields": [
+            |{"name": "customer", "type": ["string","null"]},
+            |{"name": "feature1", "type": "int"},
+            |{"name": "feature2", "type": "float"},
+            |{"name": "eventTimestamp", "type": [{"type": "long", "logicalType": "timestamp-micros"}, "null"]}
```

Comment:
Generate data creates the event timestamp as datetime, but the type is defined as long here. Seems like this is what causing the e2e test to fail.